cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreHttpClient.java

```diff
@@ -110,6 +110,18 @@ protected String getUserId() {
     return null;
   }
 
+  @Nullable
+  protected String getUserPrincipal() {
```

Comment:
This is not very good design. A better one is to have `protected void addExtraHeaders()` and by default is a no-op. It's much more flexible to have different headers added by different client.

Also, really try hard to not using `null`. It is really evil.